inc/class-saml.php

```diff
@@ -899,6 +909,7 @@ public function associateUser( $net_id, $email ) {
 				return;
 			}
 		}
+		$net_id = ! $net_id ? $username : $net_id;
```

Comment:
$net_id could be false. In that case we'll take the username to link the account.